grpc_server/server.go

```diff
@@ -0,0 +1,105 @@
+package grpc_server
+
+import (
+	"crypto/tls"
+	"net"
+	"os"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"log"
+	"reflect"
+)
+
+type grpcServerRunner struct {
+	listenAddress   string
+	handler         interface{}
+	serverRegistrar interface{}
+	tlsConfig       *tls.Config
+}
+
+// NewGRPCServer returns an ifrit.Runner for your GRPC server process, given artifacts normally generated from a
+// protobuf service definition by protoc.
+//
+// tlsConfig is optional.  If nil the server will run insecure.
+//
+// handler must be an implementation of the interface generated by protoc.
+//
+// serverRegistrar must be the "RegisterXXXServer" method generated by protoc.
+//
+// Type checking occurs at runtime.  Poorly typed `handler` or `serverRegistrar` parameters will cause a panic.
+func NewGRPCServer(listenAddress string, tlsConfig *tls.Config, handler, serverRegistrar interface{}) grpcServerRunner {
```

Comment:
Since `grpcServerRunner` is a private type, ifrit.Runner would be a better return signature.

Also, pointer vs value question. Right now grpcServerRunner is used as a value. I don't see any harm in that but generally I think it would be a pointer. 🤷‍♂️